wgpu/src/backend/web.rs

```diff
@@ -20,6 +47,38 @@ use wasm_bindgen::{prelude::*, JsCast};
 // type is (for now) harmless.  Eventually wasm32 will support threading, and depending on how this
 // is integrated (or not integrated) with values like those in webgpu, this may become unsound.
 
+fn wasm_from_id<T: FromWasmAbi>(id: ObjectId) -> T
+where
+    T::Abi: From<u32>,
+{
+    let raw = NonZeroU128::from(id);
+    let abi = <T::Abi>::from(raw.get() as u32);
+    unsafe { T::from_abi(abi) }
+}
```

Comment:
This is almost certainly unsound and needs to be unsafe too, which makes the other `From` unsound.

We do violate this principal in the rest of the codebase, so this may be worth it for the ergonomics increase.

Reply:
I've hardened the conversions by using `strict_assert` to ensure ObjectIds get converted safely, but this is still unsafe if the id was not created by something extending JsValue.